File: compile.py
import subprocess
import os
import shutil
from pathlib import Path

mainclass = input("Enter project name (no extension, i.e. 'project' not 'project.kt'): ")

directory = str(Path(__file__).parent)
temp_directory = directory + "/temp/"
jarlist = []
os.makedirs(temp_directory, exist_ok=True)

for jarfile in os.listdir(directory):
	if "jar" in str(jarfile):
		jarlist.append(jarfile)

counter = 0
classpath_string = ""

for jar in jarlist:
	classpath_string += jar+":"
	counter += 1
	if counter == len(jarlist):
		classpath_string += jar

with open("classpath",'w') as cp:
	cp.write(classpath_string)
cp.close()

compile_it = ["kotlinc", mainclass+".kt", "-include-runtime","-d",mainclass+".jar ", "-cp",classpath_string]
subprocess.run(compile_it, shell=False)

unzip_it = ["unzip","-qq",mainclass+".jar","-d "+temp_directory]
subprocess.run(unzip_it, shell=False)
exit()
project_mainclass = ""

for mainclass in os.listdir(temp_directory):
	if "class" in str(mainclass):
		project_mainclass = mainclass.replace(".class","")

shutil.rmtree(temp_directory)

run_it = ["java","-classpath",mainclass+".jar:",classpath_string, project_mainclass]
subprocess.run(run_it,shell=False)

File: compile.kt
import java.io.File
import java.nio.file.Paths
import java.nio.file.Files

fun main(){
	print("Enter project name (no extension, i.e. 'project' not 'project.kt'): ")
	var projectName:String? = readLine()
	var currentDirectory = Paths.get("").toAbsolutePath().toString()
	var dependenciesDirectory = currentDirectory + "/libs/"
	var tempDirectory = currentDirectory + "/temp/"
	var jarList:List<String> = emptyList()
	
	File(tempDirectory).mkdir()
	File(dependenciesDirectory).walk().forEach {
		if (".jar" in it.toString()){
			jarList = jarList.plus(it.toString())
		}
	}
	
	var counter:Int = 0
	var classPath:String = ""
	for (jar in jarList){
		classPath = classPath.plus(jar + ":")
		++ counter
		if (counter == jarList.size){
			classPath = classPath.plus(jar)	
		}
	}
	File("dependencies_list").writeText(classPath)
	val compileIt = listOf("kotlinc", projectName+".kt", "-include-runtime","-d",projectName+".jar", "-cp",classPath)
	ProcessBuilder(compileIt).redirectError(ProcessBuilder.Redirect.INHERIT).start().waitFor()

	val unzipIt = listOf("unzip", "-qq", currentDirectory+"/"+projectName+".jar", "-d", tempDirectory)
	ProcessBuilder(unzipIt).redirectError(ProcessBuilder.Redirect.INHERIT).start().waitFor()
	var fileList:List<String> = emptyList()
	File(tempDirectory).walk().forEach{
		if (".class" in it.toString()){
			fileList = fileList.plus(it.toString())
		}
	}
	File(tempDirectory).deleteRecursively()
	var projectMainclass:String = fileList[0].split("/")[7].replace(".class","")
	val runIt = listOf("java","-classpath",projectName+".jar:"+classPath, projectMainclass)
	ProcessBuilder(runIt).redirectOutput(ProcessBuilder.Redirect.INHERIT).redirectError(ProcessBuilder.Redirect.INHERIT).start()
}
